### PR TITLE
feat: Added props intermission and count

### PR DESCRIPTION
**Pull request by afsal-ali**
- [x] Added props intermission and count
- [x] Fixed default options issue.

### CODE REVIEW

File: packages/common/src/dotlottie-player.ts [hunk 1]
```diff
@@ -83,6 +92,16 @@ export class DotLottiePlayer {
...
 
+  protected _extraOptions: ExtraOptions;
+
+  protected _count: number = 0;
+
+  protected _counter: number = 0;
+
+  protected _intermission: number = 0;
+
+  protected _counterInterv: number | null = null;
```
Comment (theashraf) [inline packages/common/src/dotlottie-player.ts:103]:
```suggestion
  protected _counterInterval: number | null = null;
```

File: packages/common/src/dotlottie-player.ts [hunk 2]
```diff
@@ -332,13 +397,38 @@ export class DotLottiePlayer {
...
+          return;
+        }
+
+        this._counterInterv = setTimeout(() => {
+          if (!this._lottie) return;
+
+          let newDirection = this._lottie.playDirection;
+
+          if (this._mode === PlayMode.Bounce) {
+            newDirection = Number(newDirection) * -1;
```
Comment (theashraf) [inline packages/common/src/dotlottie-player.ts:420]: should we validate if the `newDirection` is a valid number first ?

File: packages/common/src/dotlottie-player.ts [hunk 3]
```diff
@@ -64,12 +64,21 @@ export interface DotLottieElement extends Element {
   __lottie?: AnimationItem | null;
 }
 
-export type RendererSettings = SVGRendererConfig & CanvasRendererConfig & HTMLRendererConfig;
-export type DotLottieConfig<T extends RendererType> = Omit<AnimationConfig<T>, 'container'> & {
-  mode?: PlayMode;
-  testId?: string | undefined;
+export const EXTRA_OPTIONS = {
+  count: 1,
+  direction: 1 as 1 | -1,
```
Comment (theashraf) [inline packages/common/src/dotlottie-player.ts:69]: i wonder why do we need this type assertion ? can't we set a type for the `EXTRA_OPTIONS`

File: packages/common/src/dotlottie-player.ts [hunk 4]
```diff
@@ -137,6 +159,17 @@ export class DotLottiePlayer {
...
 
+  protected _extractExtraOptions(config: DotLottieConfig<RendererType>): ExtraOptions {
+    return Object.entries(config).reduce((newConfig, [key, value]) => {
+      if (!Object.hasOwn(EXTRA_OPTIONS, key)) return newConfig;
+
+      return {
+        ...newConfig,
+        [key]: value,
+      };
+    }, EXTRA_OPTIONS);
```
Comment (theashraf) [inline packages/common/src/dotlottie-player.ts:170]: it needs extra cognitive load to process the purpose of using `reduce` in this function, i wonder if it's possible to use normal `forEach` or `for...of` to improve the readability of the code. what do you think ?

```suggestion
   const extraOptions: Partial<ExtraOptions> = {};
  
  for (const [key, value] of Object.entries(config)) {
    ....
  }

  return extraOptions
```

File: packages/common/src/dotlottie-player.ts [hunk 5]
```diff
@@ -116,19 +135,22 @@ export class DotLottiePlayer {
...
-
     if (options?.testId) {
       this._testId = options.testId;
     }
 
+    this._extraOptions = this._extractExtraOptions(options || ({} as DotLottieConfig<RendererType>));
+
+    this.setCount(this._extraOptions.count);
+    this.setIntermission(this._extraOptions.intermission);
+    this.setMode(this._extraOptions.mode);
```
Comment (theashraf) [inline packages/common/src/dotlottie-player.ts:146]: do we have runtime validation for these properties ?

Comment (afsal-ali) [reply packages/common/src/dotlottie-player.ts:154]: added validation to `_extractExtraOptions` function

**Pull request merged**